es_ES/widgets-intro.md

```diff
@@ -0,0 +1,799 @@
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(5.0),
+          color: Colors.lightGreen[500],
+        ),
+        child: Center(
+          child: Text('Contratar'),
+        ),
+      ),
+    );
+  }
+}
+```
+
+El widget
+[`GestureDetector`](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html)
+no tiene una representación visual sino que detecta los gestos realizados por el usuario.
+Cuando el usuario pulsa en
+[`Container`](https://docs.flutter.io/flutter/widgets/Container-class.html),
+el
+[`GestureDetector`](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html)
+llama a su callback
+[`onTap`](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html#onTap)
+en este caso imprimiendo un mensaje a la consola. Puede utilizar
+[`GestureDetector`](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html)
+para detectar una variedad de gestos de entrada, incluidos taps, drags y escalas.
+
+Muchos widgets usan un
+[`GestureDetector`](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html)
+para proporcionar callbacks opcionales para otros widgets. Por ejemplo, los widgets
+[`IconButton`](https://docs.flutter.io/flutter/material/IconButton-class.html),
+[`RaisedButton`](https://docs.flutter.io/flutter/material/RaisedButton-class.html),
+y
+[`FloatingActionButton`](https://docs.flutter.io/flutter/material/FloatingActionButton-class.html)
+tienen
+[`onPressed`](https://docs.flutter.io/flutter/material/RaisedButton-class.html#onPressed)
+callbacks que se activan cuando el usuario toca el widget.
+
+Cambiar los widgets en respuesta a las entradas
+-----------------------------------------------
+
+_Artítulos principales: [`StatefulWidget`](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html), [`State.setState`](https://docs.flutter.io/flutter/widgets/State/setState.html)_
+
+Hasta ahora, solo hemos usado widgets sin estado. Los widgets sin estado
+reciben argumentos de su widget padre, que almacenan en variables de miembro
+[`final`](https://www.dartlang.org/docs/dart-up-and-running/ch02.html#final-and-const).
+Cuando se le pide a un widget que se
+[`construya`](https://docs.flutter.io/flutter/widgets/StatelessWidget/build.html),
+usa estos valores almacenados para derivar nuevos argumentos para los widgets que crea.
+
+Para construir experiencias mas complejas&mdash;por ejemplo, reaccionar de forma
+más interesante a las aplicaciones&mdash;de entrada de usuario que típicamente llevan algún estado.
+Flutter utiliza StatefulWidgets para capturar esta idea. StatefulWidgets son
+widgets especiales que saben como generar objetos State, que luego se usan
+para mantener el estado. Considera este ejemplo básico, usando el
+[`RaisedButton`](https://docs.flutter.io/flutter/material/RaisedButton-class.html)
+mencionado anteriormente:
+
+```dart
+class Counter extends StatefulWidget {
+  // Esta clase es la configuración para el estado. Tiene
```

Comment:
//It holds the
  // values (in this nothing) provided by the parent and used by the build
  // method of the State.

Guarda los valores (nada en este ejemplo) proporcionados por el padre y usados por el método build del State,

Reply:
Listo corregido